packages/svelte2tsx/src/svelte2tsx/processInstanceScriptContent.ts

```diff
@@ -385,3 +394,33 @@ export function processInstanceScriptContent(
         generics
     };
 }
+
+function transformInterfacesToTypes(tsAst: ts.SourceFile, str: MagicString, astOffset: any) {
+    tsAst.statements.filter(ts.isInterfaceDeclaration).forEach((node) => {
+        str.overwrite(
+            node.getStart() + astOffset,
+            node.getStart() + astOffset + 'interface'.length,
+            'type'
+        );
+
+        if (node.heritageClauses?.length) {
+            const extendsStart = str.original.indexOf('extends', node.getStart() + astOffset);
```

Comment:
The interface might also be generic and has generic constraints. I found we can use `node.heritageClauses.pos` as the starting point instead.

Reply:
Oh yeah totally forgot about that, good catch. I'll adjust and add a test.